src/librustc_trans/save/dump_csv.rs

```diff
@@ -814,10 +820,60 @@ impl <'l, 'tcx> DumpCsvVisitor<'l, 'tcx> {
                                   &typ);
         }
     }
+
+    /// If span is generated code, extract the source callsite and callee spans.
+    ///
+    /// If there is no entry in the mac_uses map for the callee span, write out a macro
+    /// def and macro use. Otherwise write out a macro use if the callsite span isn't already
+    /// mapped to the callee span.
+    fn process_generated(&mut self, span: Span, id: NodeId) {
+        let data = self.save_ctxt.get_macro_use_data(span, id);
+        if let None = data {
+            return;
+        }
+        let data = data.unwrap();
+        if let Some(mac) = self.sess.imported_macros.borrow().get(&data.callee_span) {
+            let &(ref mac_name, mac_span) = mac;
+            let data = MacroUseData { name: mac_name.clone(),
+                                      callee_span: mac_span,
+                                      .. data };
+            self.process_macro_use(data, false);
```

Comment:
That's a fair point. In which case I can probably remove process_generated entirely, move the checks into get_macro_use and the rest into process_macro_use